src/compiler/checker.ts

```diff
@@ -27132,6 +27132,23 @@ namespace ts {
             }
         }
 
+        function checkGrammarForDuplicateIdentifier(properties: ReadonlyArray<ObjectLiteralElement>, spread: SpreadAssignment, spreadIndex: number) {
+            const type = getTypeOfExpression(spread.expression);
+            if (type && type.flags & TypeFlags.Object) {
+                const resolvedObjectType = resolveStructuredTypeMembers(type as StructuredType);
+                for (let i = 0; i < spreadIndex; ++i) {
+                    const prop = properties[i];
+                    if (!isSpreadAssignment(prop)) {
+                        const propName = getTextOfPropertyName(prop.name);

```

Comment:
`propName` may be undefined. that could be handled in this PR so there's no need to update it in the `--strictNullChecks` branch.